### PR TITLE
Martin/ios 523 message status bug

### PR DESCRIPTION
**Pull request by MartinJibber**

### CODE REVIEW

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 1]
```diff
@@ -320,12 +320,12 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
 
                 let message = ChatClient.shared.message(cid: cid, id: messageID)
 
-                if !message.isFromCurrentUser && state.detailShown {
+                if !message.isFromCurrentUser && state.areDetailsFullyVisible {
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift:323]: Making it more clear what this bool means.

File: iOS/Flows/Conversation/Cells/MessageCell.swift [hunk 2]
```diff
@@ -117,69 +119,66 @@ class MessageCell: UICollectionViewCell {
...
+
+        self.handleDetailVisibility(areDetailsFullyVisible: areDetailsFullyVisible)
     }
 
-    // MARK: - Message Consumption
+    // MARK: - Message Detail Tasks
 
-    private var consumeMessageTask: Task<Void, Never>?
-    private var statusTextTask: Task<Void, Never>?
+    private var messageDetailTasks = TaskPool()
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/MessageCell.swift:132]: Putting the detail tasks together into a task pool.

File: iOS/Flows/Conversation/Cells/MessageCell.swift [hunk 3]
```diff
@@ -117,69 +119,66 @@ class MessageCell: UICollectionViewCell {
...
+        guard areDetailsFullyVisible, ChatUser.currentUserRole != .anonymous else { return }
 
         let message = ChatClient.shared.message(cid: cid, id: messageable.id)
 
-        self.startConsumptionIfNeeded(for: message)
-        self.updateStatusText(for: message)
+        // Show the time sent text if needed.
+        if message.lastUpdatedAt?.getTimeAgoString() != self.messageState.statusText {
+            self.startTimeSentTextTask(for: message)
+        }
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/MessageCell.swift:150]: Only change the status text if it's not already showing the time.

File: iOS/Flows/Conversation/Cells/MessageCell.swift [hunk 4]
```diff
@@ -87,11 +86,14 @@ class MessageCell: UICollectionViewCell {
...
     }
 
+    // MARK: Configuration
+
     func configure(with message: Messageable) {
         self.content.configure(with: message)
         
         self.messageState.message = message
         self.messageState.deliveryStatus = message.deliveryStatus
+        self.messageState.statusText = message.context.displayName
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/MessageCell.swift:96]: Make sure the status text is configured when a new message is assigned for this cell. This fixes cell recycling issues.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**